### PR TITLE
test(r/rbac): improve test coverage with comprehensive edge cases

### PR DESCRIPTION
**Pull request by junghoon-vans**
## Summary
- Add comprehensive edge case tests for ownership and validation functions
- Improve TestIsPendingOwner with actual pending owner scenarios
- Add extensive edge case coverage for all RBAC operations

## Test Coverage Improvements
- **Private method coverage**: Added tests for all private validation functions
- **Ownership edge cases**: Invalid/malformed addresses, self-transfer, multiple transfers, accept without pending
- **Authorization tests**: Comprehensive admin/governance/unauthorized caller scenarios
- **Boundary validation**: Empty strings, invalid formats, duplicate operations

### Enhanced Files
- **assert_test.gno**
  - Added `TestAssert_assertIsValidRoleName` (3 cases)
  - Added `TestAssert_assertIsValidAddress` (5 cases)
  - Added `TestAssert_assertIsAdminOrGovernance` (5 cases)

- **ownership_test.gno**
  - Extended `TestOwnership_TransferOwnership` with 3 edge cases
  - Added `TestOwnership_TransferOwnershipTwice` for pending owner update validation
  - Added `TestOwnership_AcceptOwnershipWithoutPendingOwner` for error handling
  - Improved `TestIsPendingOwner` from 3 to 7 cases

- **rbac_test.gno**
  - Added comprehensive authorization tests for RegisterRole, UpdateRoleAddress, RemoveRole
  - Added extensive test cases covering admin, governance, and unauthorized callers
  - Added validation tests for empty/invalid inputs
  - Added integration tests for RBAC-to-Access sync

## Notes
- Removed emit_test.gno: Gno currently lacks ability to verify events within tests

### CODE REVIEW

File: contract/r/gnoswap/rbac/emit_test.gno [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+			// Setup code realm context so runtime.PreviousRealm() works
+			testing.SetRealm(testing.NewCodeRealm("gno.land/r/gnoswap/rbac"))
+
+			// Should not panic - just emits event
+			emitRemoveRoleEvent(tt.roleName)
+		})
+	}
+}
+
+func TestEmit_emitUpdateRoleAddressEvent(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (junghoon-vans) [reply contract/r/gnoswap/rbac/emit_test.gno]: 2488759
